src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -55,186 +56,199 @@ public bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext co
+                        //      /*
+                        //      |*/
+                        return commentIndentation + " ";
+                    }
+                    else if (caretPosition == firstNonWhitespacePosition + 1)
+                    {
+                        //      *|/
+                        return lineIndentation + "*";
+                    }
+                    else
+                    {
+                        //      /*
+                        //   |   */
+                        return commentIndentation + " ";
+                    }
                 }
-                else
+
+                if (startsWithBlockCommentMiddleString)
                 {
-                    //      /*
-                    //   |   */
-                    return " * ";
+                    if (BlockCommentEndsRightAfterCaret(caretPosition))
+                    {
+                        //      *|*/
+                        return lineIndentation;
+                    }
+                    else if (caretPosition > firstNonWhitespacePosition)
+                    {
+                        //      *|
+                        return lineIndentation + "*" + GetPaddingAfterCommentCharacter();
+                    }
+                    else
+                    {
+                        //      /*
+                        //   |   *
+                        return commentIndentation + " ";
+                    }
                 }
+
+                return null;
             }
 
-            if (currentLineStartsWithBlockCommentMiddleString)
+            string GetPaddingAfterCommentCharacter()
             {
-                if (BlockCommentEndsRightAfterCaret(caretPosition))
-                {
-                    //      *|*/
-                    return "";
-                }
-                else if (caretPosition > firstNonWhitespacePosition)
-                {
-                    //      *|
-                    return "*" + GetPaddingOrIndentation(currentLine, caretPosition, firstNonWhitespacePosition, "*");
-                }
-                else
-                {
-                    //      /*
-                    //   |   *
-                    return " * ";
-                }
-            }
+                var currentChar = firstNonWhitespacePosition;
+                Debug.Assert(textSnapshot[currentChar] == '/' || textSnapshot[currentChar] == '*');
+
+                // Skip past the first comment char.
+                currentChar++;
 
-            return null;
+                // Skip past any banner of ****'s 
+                while (currentChar < caretPosition && textSnapshot[currentChar] == '*')
+                    currentChar++;
+
+                var start = currentChar;
+                while (currentChar < caretPosition && char.IsWhiteSpace(textSnapshot[currentChar]))
+                    currentChar++;
+
+                return textSnapshot.GetText(Span.FromBounds(start, currentChar));
+            }
```

Comment:
imo, the new logic is much saner and easier to undertand.

Reply:
will add.

Reply:
added docs and renamed to be much clearer: GetWhitespaceBetweenCommentAsteriskAndCaret